Tasks/AppCenterDistributeV3/appcenterdistribute.ts

```diff
@@ -0,0 +1,511 @@
+    tl.debug("-- Finishing uploading release...");
+    let defer = Q.defer<string>();
+    let commitReleaseUrl: string = `${apiServer}/${apiVersion}/apps/${appSlug}/release_uploads/${upload_id}`;
+    tl.debug(`---- url: ${commitReleaseUrl}`);
+    let headers = {
+        "X-API-Token": token,
+        "User-Agent": userAgent,
+        "internal-request-source": "VSTS"
+    };
+
+    let commitBody = { "status": "committed" };
+
+    request.patch({ url: commitReleaseUrl, headers: headers, json: commitBody }, (err, res, body) => {
+        responseHandler(defer, err, res, body, () => {
+            if (body && body['release_url']) {
+                defer.resolve(body['release_url']);
+            } else {
+                defer.reject(tl.loc("FailedToUploadFile"));
+            }
+        });
+    })
+
+    return defer.promise;
+}
+
+function publishRelease(apiServer: string, releaseUrl: string, isMandatory: boolean, isSilent: boolean, releaseNotes: string, destinationIds: string[], token: string, userAgent: string) {
+    tl.debug("-- Mark package available.");
+    let defer = Q.defer<void>();
+    let publishReleaseUrl: string = `${apiServer}/${releaseUrl}`;
+    tl.debug(`---- url: ${publishReleaseUrl}`);
+
+    let headers = {
+        "X-API-Token": token,
+        "User-Agent": userAgent,
+        "internal-request-source": "VSTS"
+    };
+    const destinations = destinationIds.map(id => { return { "id": id }; });
+    let publishBody = {
+        "status": "available",
+        "release_notes": releaseNotes,
+        "mandatory_update": isMandatory,
+        "destinations": destinations
+    };
+
+    if (isSilent) {
+        publishBody["notify_testers"] = false;
+    }
+
+    let branchName = process.env['BUILD_SOURCEBRANCH'];
+    branchName = getBranchName(branchName);
+    const sourceVersion = process.env['BUILD_SOURCEVERSION'];
+    const buildId = process.env['BUILD_BUILDID'];
+
+    // Builds started by App Center has the commit message set when distribution is enabled
+    const commitMessage = process.env['LASTCOMMITMESSAGE'];
+    // Updating the internal_request_source to distinguish the AppCenter triggered build and custom build
+    if (!!commitMessage) {
+        headers["internal-request-source"] = "VSTS-APPCENTER";
+    }
+
+    // Including these information for distribution notification to have additional context
+    // Commit message is optional
+    if (branchName && sourceVersion) {
+        const build = {
+            id: buildId,
+            branch: branchName,
+            commit_hash: sourceVersion
+        }
+
+        if (commitMessage) {
+            build['commit_message'] = commitMessage;
+        }
+
+        publishBody = Object.assign(publishBody, { build: build });
+    }
+
+    request.patch({ url: publishReleaseUrl, headers: headers, json: publishBody }, (err, res, body) => {
```

Comment:
Is this now using the new Distribution API (which differentiates between group, store and tester), or is this still the old one? For V3, now that we have destination type, we should move to the new one.

Reply:
No, it still uses old api. Thanks for note, I will switch to new one

Reply:
Updated to use new post api for publishing to destination (1 api call per destination)
Also added additional api call to set release notes, build etc